tasks/users.yml

```diff
@@ -0,0 +1,64 @@
+---
+- block:
+    - name: Create tmpfile
+      tempfile:
+        state: file
+      register: tmpfile
+      changed_when: false
+      tags:
+        - samba
+        - samba_user
+
+    - name: Ensure more secure file permissions
+      file:
+        path: "{{ tmpfile.path }}"
+        mode: "0600"
+      tags:
+        - samba
+        - samba_user
+
+    - name: Store password in tmpfile
+      copy:
+        content: |
+          {{ user.password }}
+          {{ user.password }}
```

Comment:
Is this intentionally the same thing twice?

Reply:
Yes. As you can see from the shell command later, the contents of this file are just catted into smbpasswd. smbpasswd asks for the password and then password confirmation right after, so we need to enter it twice.

Reply:
Might be worth adding a comment to that effect to prevent someone "fixing" it in the future :)